src/htmlgen.cpp

```diff
@@ -2666,15 +2673,22 @@ void HtmlGenerator::writeSearchPage()
   QFile f(fileName);
   if (f.open(IO_WriteOnly))
   {
+    QCString htmlFileExtension=Config_getString(HTML_FILE_EXTENSION);
+    htmlFileExtension=htmlFileExtension.stripWhiteSpace();
+    if (htmlFileExtension.isEmpty())
+    {
+      htmlFileExtension = ".html";
+    }
+
```

Comment:
I don't think this correction of the extension is necessary as in `configimpl.l` we see:
```
  // check & correct HTML_FILE_EXTENSION
  QCString htmlFileExtension=Config_getString(HTML_FILE_EXTENSION);
  htmlFileExtension=htmlFileExtension.stripWhiteSpace();
  if (htmlFileExtension.isEmpty())
  {
    htmlFileExtension = ".html";
  }
  Config_updateString(HTML_FILE_EXTENSION,htmlFileExtension);
```
so the correction is done and and the corrected value  is placed back in `HTML_FILE_EXTENSION` so that subsequent calls to `Config_getString(HTML_FILE_EXTENSION)` directly return the corrected value.

Reply:
I found this correction in other parts of the code; not knowing the project I simply replicated it.
Doxygen::htmlFileExtension does the job and we get much cleaner code